File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {

    map<string, int> m;
    
    for(string p : participant)
        m[p]++;
    for(string p : completion)
        m[p]--;
    
    string ret = "";
    for(string p : participant)
        if(m[p] > 0)
            ret = p;
    
    return ret;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        
        Map<String, Integer> map = Arrays.stream(completion)
            .collect(Collectors.toMap(String::valueOf, (s) -> 1, (o, n) -> o + n));
        
        String answer = Arrays.stream(participant)
            .filter(i -> {
                if(!map.containsKey(i)){
                    return true;
                }
                int v = map.get(i);
                if(v - 1 == 0){
                    map.remove(i);
                }else{
                    map.put(i, v - 1);
                }
                return false;
            })
            .findFirst().get();
            
        return answer;
    }
}